tests/integration/test_s3_latency_log/test.py

```diff
@@ -0,0 +1,208 @@
+
+    node.query("INSERT INTO s3_latency_log_test VALUES (1, 'Hello')")
+
+    node.query("SYSTEM FLUSH LOGS")
+    latency_log_after = literal_eval(
+        node.query(
+            """
+        SELECT (LatencyEvent_S3WriteConnectMicroseconds, LatencyEvent_S3FirstByteWriteAttempt1Microseconds, LatencyEvent_S3FirstByteWriteAttempt2Microseconds, LatencyEvent_S3FirstByteWriteAttemptNMicroseconds)
+        FROM system.latency_log
+        ORDER BY event_time_microseconds DESC LIMIT 1
+        """
+        )
+    )
+    connect_times_after = literal_eval(
+        node.query(
+            """
+        WITH (SELECT LatencyEvent_DiskS3WriteConnectMicroseconds FROM system.latency_buckets) AS buckets
+        SELECT format('[{}]', arrayStringConcat(arrayMap((t, c) ->
+            format('({},{})', toString(t), c), buckets, LatencyEvent_DiskS3WriteConnectMicroseconds), ','))
+        FROM system.latency_log
+        ORDER BY event_time_microseconds DESC LIMIT 1
+        """
+        )
+    )
+    connect_timed_out = (
+        int(
+            node.count_in_log(
+                f"Failed to make request to.*{RESOLVER_CONTAINER_NAME}:{RESOLVER_PORT}.*Timeout: connect timed out"
+            )
+        )
+        - connect_timed_out
+    )
+    resolver_requests_cnt = (
+        int(
+            cluster.exec_in_container(
+                cluster.get_container_id("resolver"),
+                [
+                    "curl",
+                    "-s",
+                    f"http://{RESOLVER_CONTAINER_NAME}:{RESOLVER_PORT}/total",
+                ],
+                nothrow=True,
+            )
+        )
+        - resolver_requests_cnt
+    )
+    logging.debug("Latency log after - %s", latency_log_after)
+    logging.debug("Connect times after - %s", connect_times_after)
+    logging.debug("Connection timed out requests count after - %d", connect_timed_out)
+    logging.debug("Resolver requests count after - %d", resolver_requests_cnt)
+
+    for idx_event, event in enumerate(latency_log_after):
+        prev = 0
+
+        for i, cnt in enumerate(event):
+            curr_cnt = cnt - latency_log_before[idx_event][i]
+            assert (
+                curr_cnt >= prev
+            ), "All counters in latency log should be in non-decreasing order"
+            prev = curr_cnt
+
+    below_second_connect_time_cnt = 0
+    other_connect_time_cnt = 0
+
+    # Count connect times that are below connect timeout, and then we check that
+    # for requests with connection timeout error we have correct latency log entries.
+    # Also we expect no connection timeout for requests to minio.
+    for i, (bucket, cnt) in enumerate(connect_times_after):
+        cnt_before = connect_times_before[i][1]
+
+        if bucket < CONNECT_TIMEOUT_MICROSECONDS:
+            below_second_connect_time_cnt = cnt - cnt_before
+        else:
+            other_connect_time_cnt = cnt - cnt_before
+            break
+
+    logging.debug(
```

Comment:
Just curios. Do you know how to find this logs after a fail?
I do not know.

Reply:
There's a `pytest.log` in `tests/integration` directory when you run the test